operator/pkg/ciliumendpointslice/reconciler.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ciliumendpointslice
+
+import (
+	"context"
+
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	clientset "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// reconciler is used to sync the current (i.e. desired) state of the CESs in datastore into current state CESs in the k8s-apiserver.
+// The source of truth is in local datastore.
+type reconciler struct {
+	client     clientset.CiliumV2alpha1Interface
+	cesManager operations
+}
+
+// newReconciler creates and initializes a new reconciler.
+func newReconciler(client clientset.CiliumV2alpha1Interface, cesMgr operations) *reconciler {
+	return &reconciler{
+		client:     client,
+		cesManager: cesMgr,
+	}
+}
+
+// Create a new CES in api-server.
+func (r *reconciler) reconcileCESCreate(cesToCreate string) (err error) {
+	var retCES, ces *cilium_v2.CiliumEndpointSlice
+	// Get the copy of CES from the cesManager
+	if ces, err = r.cesManager.getCESCopyFromCache(cesToCreate); err != nil {
+		return
+	}
+
+	// Call the client API, to Create CES
+	if retCES, err = r.client.CiliumEndpointSlices().Create(
+		context.TODO(), ces, metav1.CreateOptions{}); err != nil {
+		log.WithError(err).WithFields(logrus.Fields{
+			logfields.CESName: cesToCreate,
+		}).Info("Unable to create CiliumEndpointSlice in k8s-apiserver")
```

Comment:
This should be error instead of info? (this applies to other occurrences in this file too)

Reply:
Its quite normal to see failure from api-server for various reasons, hence logging this message as Info is sufficient.  As well, CI tests fail if it see any errors on operator logs.

Reply:
@krishgobinath what is the impact in that case? Will the logic retry or does it give up? We could potentially add "Retrying..." if that's the case, or otherwise if this is lossy (no retries), maybe it should be a warning or error.

Reply:
@joestringer Yes we do have decent retry logic in our code, All these errors are `retried`  for certain number of times, with backoff time. Eventually if all `retried` failed, this particular request dropped out of workqueue and logs an `error`.

https://github.com/cilium/cilium/pull/17658/files#diff-a573d3ad970b6a75e585561721687ef8942bd4d84cff02ce1e283591421b2261R254-R275